### PR TITLE
Fixes production mode in Autotools and Linux compiler flags

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

File: config/linux-gnu [hunk 1]
```diff
@@ -208,14 +208,15 @@ fi
 
 case $CC_BASENAME in
   gcc)
+    CFLAGS="-Wall -Wcast-qual -Wconversion -Wextra -Wfloat-equal -Wformat=2 -Winit-self -Winvalid-pch -Wmissing-include-dirs -Wshadow -Wundef -Wwrite-strings -pedantic -Wno-c++-compat"
```
Comment (derobins) [inline config/linux-gnu:211]: These are the 'general' warning flags from HDF5.

File: configure.ac [hunk 2]
```diff
@@ -860,7 +860,7 @@ AC_ARG_ENABLE([production],
               [AS_HELP_STRING([--enable-production],
                               [Determines how to run the compiler.])])
 
-case "X-$enable-production" in
+case "X-$enable_production" in
```
Comment (derobins) [inline configure.ac:863]: The actual value checked should be $enableval or $enable_<feature>, with an underscore, not a hyphen.

File: libhdf4.settings.in [hunk 3]
```diff
@@ -14,17 +14,16 @@ General Information:
...
                Shared Libraries: @enable_shared@
                Static Libraries: @enable_static@
   Statically Linked Executables: @STATIC_EXEC@
                         LDFLAGS: @LDFLAGS@
- 	 	Extra libraries: @LIBS@
- 		       Archiver: @AR@
- 		 	 Ranlib: @RANLIB@
+                Extra libraries: @LIBS@
+                       Archiver: @AR@
+                         Ranlib: @RANLIB@
```
Comment (derobins) [inline libhdf4.settings.in:26]: These were just misaligned.

File: libhdf4.settings.in [hunk 4]
```diff
@@ -14,17 +14,16 @@ General Information:
 
 Compiling Options:
 ------------------
-               Compilation Mode: @CONFIG_MODE@
```
Comment (derobins) [inline libhdf4.settings.in:17]: This is duplicated in the block above, so I removed it here.

File: configure.ac [hunk 5]
```diff
@@ -860,7 +860,7 @@ AC_ARG_ENABLE([production],
               [AS_HELP_STRING([--enable-production],
                               [Determines how to run the compiler.])])
```
Comment (lrknox) [inline configure.ac:862]: Don't you need ", [enable_production $enableval]" following "[Determines how to run the compiler.])]"?

Comment (derobins) [reply configure.ac:862]: Nope. The default behavior suffices.

Comment (derobins) [reply configure.ac:862]: If you look at the configure output in the CI for this PR, it's still broken, but that's because setting CC is broken.

**Review by lrknox — APPROVED** [review]

**Pull request merged**